tests/export_back_compat_test.py

```diff
@@ -301,22 +305,24 @@ def test_cpu_eigh_lapack_syevd(self, dtype_name="f32"):
       for dtype_name in ("f32", "f64")
       # We use different custom calls for sizes <= 32
       for variant in ["syevj", "syevd"])
-  def test_cuda_eigh_cusolver_syev(self, dtype_name="f32", variant="syevj"):
+  def test_gpu_eigh_solver_syev(self, dtype_name="f32", variant="syevj"):
     if not config.enable_x64.value and dtype_name == "f64":
       self.skipTest("Test disabled for x32 mode")
-    if (jtu.test_device_matches(["cuda"]) and
```

Comment:
I was not aware of this conditional about the cusolver version. I think this points to a backwards compatibility failure here, albeit not one in the current PR. We queried the workspace size when we lowered the code, and we expect this code to be runnable with a new version of cusolver.

@dfm @pparuzel FYI, since you are thinking about how to handle the workspace size issues when we lower for GPU.

Reply:
Thanks! Yeah this issue should resolve itself once I finish porting the kernels to the FFI using the `ScratchAllocator`. See here for an example: https://github.com/google/jax/pull/22795